From ce20d0346d9982a10fd0076b53e3e3296c649b47 Mon Sep 17 00:00:00 2001
From: Larry Finger <Larry.Finger@lwfinger.net>
Date: Sun, 23 Mar 2014 00:22:46 -0500
Subject: [PATCH 457/458] rtl8192du: Fix firmware problem

When the driver was converted to external firmware, the fact that
only devices with a dual MAC need to have such firmware loaded.
When this firmware is loaded into single MAC devices, all manner of
problems occur.

Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
---
 hal/rtl8192d_hal_init.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
+++ b/drivers/staging/rtl8192du/hal/rtl8192d_hal_init.c
@@ -331,6 +331,10 @@ int FirmwareDownload92D(struct rtw_adapt
 	if (adapter->bSurpriseRemoved)
 		return _FAIL;
 
+	/* Single MAC Single PHY units break if external firmware is loaded */
+	if (pHalData->MacPhyMode92D == SINGLEMAC_SINGLEPHY)
+		return rtStatus;
+
 	if (!adapter->firmware || !adapter->firmware->buffer) {
 		if (!get_fw_from_file(adapter)) {
 			rtStatus = _FAIL;
